electron-app/src/python/pyrunner.py

```diff
@@ -141,23 +141,43 @@ def BuildAndRun(
                 ) as f:
                     Dockerfile = f.read()
                 with open(
-                    os.path.join(os.path.dirname(__file__), "run_docker_sh"), "r"
+                    os.path.join(os.path.dirname(__file__), "build_container_sh"), "r"
                 ) as f:
-                    docker_launcher = f.read()
+                    docker_build_container = f.read()
+                with open(
+                    os.path.join(os.path.dirname(__file__), "launch_container_sh"), "r"
+                ) as f:
+                    docker_launch_container = f.read()
                 # Write Dockerfile and launch script to zip
                 with ZipFile(zipfilename, "a") as zipfile:
+                    # Build container
                     zipfile.writestr(
                         "Dockerfile",
                         Dockerfile.replace(
                             "$(snakemake --list)",
                             " ".join(target_rules),
                         ),
                     )
-                    info = ZipInfo("run_docker.sh")
+                    # Build container
+                    info = ZipInfo("build_container.sh")
+                    info.external_attr = 0o0100777 << 16  # give rwx permissions
+                    zipfile.writestr(
+                        info,
+                        docker_build_container.replace(
+                            "#!/usr/bin/env bash", shell_launch
+                        ).replace(
+                            "$(snakemake --list)",
+                            " ".join(target_rules),
+                        ),
+                    )
+                    # Launch workflow
+                    info = ZipInfo("launch_container.sh")
                     info.external_attr = 0o0100777 << 16  # give rwx permissions
                     zipfile.writestr(
                         info,
-                        docker_launcher.replace("#!/usr/bin/env bash", shell_launch),
+                        docker_launch_container.replace(
+                            "#!/usr/bin/env bash", shell_launch
```

Comment:
is `shell_launch` a different shell?

Reply:
I added this as a method to switch shells since workflows may be run outside a container in an uncontrolled environment. This is actually important as `conda` in particular relies on initialisation via the users' shell startup script, meaning that only the user-preferred shell may be initialised to allow conda to run in a subprocess. At the moment `shell_launch` is hard-coded to bash, but the plan was to add this option to the GUI as an option when not containerising the workflow [at the moment we don't separate container vs non-container builds; though this is a useful reminder to look again at this issue]. If we decide to run _only_ containers then we can take it out.